src/main/java/seedu/tp/group/GroupList.java

```diff
@@ -63,4 +63,48 @@ public void addFlashcardToOneGroup(Ui ui, FlashcardList flashcardList) throws Hi
             throw new InvalidFlashcardIndexException();
         }
     }
+
+    public List<FlashcardGroup> getGroups() {
+        return groups;
+    }
+
+    /**
+     * Gets the flashcard group at a specific index.
+     *
+     * @param index the index.
+     * @return the flashcard group at the specified index
+     */
+    public FlashcardGroup getGroupAtIdx(int index) {
+        return groups.get(index);
+    }
+
+    /**
+     * Gets the total number of groups in the list.
+     *
+     * @return total number of groups in the list
+     */
+    public int getTotalGroupNum() {
+        return groups.size();
+    }
+
+    /**
+     * Check if the current instance is equal to the object passed in.
+     *
+     * @param obj The object to be compared against the current instance
+     * @return whether or not the two objects are equal
+     */
+    @Override
+    public boolean equals(Object obj) {
+        GroupList otherGroupList = (GroupList) obj;
+        if (this.getTotalGroupNum() != otherGroupList.getTotalGroupNum()) {
+            return false;
+        }
+
+        for (int idx = 0; idx < otherGroupList.getTotalGroupNum(); idx++) {
+            if (!this.getGroups().contains(otherGroupList.getGroupAtIdx(idx))) {
```

Comment:
We can use `!this.getGroupAtIdx(idx).equals(otherGroupList.getGroupAtIdx(idx))` if we have avoided groups with unordered equal flashcard lists in tests

Reply:
Yeah, I think we can use ```!this.getGroupAtIdx(idx).equals(otherGroupList.getGroupAtIdx(idx))``` here. But I guess this is because we avoid groupLists with unordered equal groups and it seems we have no need to do unordered equal check for grouplist.

And since I decide to do sorting before check the equality of flashcard lists. I think we have no need to avoid checking groups with unordered equal flashcard lists in test any more.